typescript/packages/well-log-viewer/src/SyncLogViewer.tsx

```diff
@@ -370,19 +380,23 @@ class SyncLogViewer extends Component<SyncLogViewerProps, State> {
         return ret;
     }
 
-    beforeRender(props: SyncLogViewerProps): void {
+    beforeRender(nextProps: SyncLogViewerProps): void {
         // called before render() but not inside it to avoid onDeleteController notifications
-        if (this.callbackManagers.length === props.welllogs.length) return;
-        this.spacers.length = props.welllogs.length;
+        this.wellLogCollections = getWellLogCollectionsFromProps(nextProps);
+
+        if (this.callbackManagers.length === this.wellLogCollections.length)
+            return;
+        this.spacers.length = this.wellLogCollections.length;
 
-        this.fillViewsCallbacks(props.welllogs.length); // update this.callbackManagers and this.callbacks[] before render()
+        this.fillViewsCallbacks(this.wellLogCollections.length); // update this.callbackManagers and this.callbacks[] before render()
     }
 
     componentDidUpdate(
         prevProps: SyncLogViewerProps /*, prevState: State, snapshot*/
     ): void {
         if (
             this.props.welllogs !== prevProps.welllogs ||
+            this.props.wellLogCollections !== prevProps.wellLogCollections ||
```

Comment:
this.props.wellLogCollections ?? this.props.welllogs !== prevProps.wellLogCollections ?? prevProps.welllogs